src/test/java/jenkins/plugins/logstash/persistence/ElasticSearchSSLCertsTest.java

```diff
@@ -0,0 +1,212 @@
+package jenkins.plugins.logstash.persistence;
+
+import jenkins.plugins.logstash.configuration.ElasticSearch;
+
+import org.apache.http.HttpRequest;
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpException;
+import org.apache.http.HttpStatus;
+import org.apache.http.protocol.HttpRequestHandler;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.conn.ssl.TrustSelfSignedStrategy;
+import org.apache.http.protocol.HttpContext;
+import org.apache.http.impl.bootstrap.HttpServer;
+import org.apache.http.impl.bootstrap.ServerBootstrap;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.ssl.SSLContextBuilder;
+import org.apache.http.ssl.SSLContexts;
+import org.hamcrest.core.IsInstanceOf;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.Inet4Address;
+import java.net.URL;
+import java.net.URI;
+import java.net.UnknownHostException;
+import java.security.KeyManagementException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLException;
+import javax.net.ssl.SSLHandshakeException;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+
+public class ElasticSearchSSLCertsTest {
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    private static final KeyStore NO_CLIENT_KEYSTORE = null;
+    private static final SSLContext NO_SSL_CONTEXT = null;
+    private static final TrustManager[] NO_SERVER_TRUST_MANAGER = null;
+
+    private static final char[] KEYPASS_AND_STOREPASS_VALUE = "aaaaaa".toCharArray();
+    private static final String JAVA_KEYSTORE = "jks";
+    private static final String CLIENT_KEYSTORE = "elasticsearch-sslcerts/keystore.ks";
+    private static final String CLIENT_TRUSTSTORE = "elasticsearch-sslcerts/truststore.ks";
+
+    @Test
+    public void httpPost_HttpServer_Returns200OK() throws Exception {
```

Comment:
minor: I think we can drop the middle part that is repeated in every test (add it to test name?)
`httpPost_Returns200OK` looks informative enough for me, even `http_Returns200OK`

Reply:
The middle part naming convention represents the nature of the server (`HttpServer` vs `Http*s*Server`). I get that the _s_ may be easy to miss, but I'm not sure if there's a better naming convention taking into account both, the POST request and the server config?

Reply:
oh, ok I've actually missed `httpsPost_HttpServer` case

maybe use "ssl/nossl" instead, like `ssl_post_nosll_server" ?

Reply:
Resolved